src/reflect/scala/reflect/internal/Definitions.scala

```diff
@@ -1358,24 +1358,25 @@ trait Definitions extends api.StandardDefinitions {
       clazz setInfo GenPolyType(List(tparam), ClassInfoType(parents, newScope, clazz)) markAllCompleted
     }
 
-    def newPolyMethod(typeParamCount: Int, owner: Symbol, name: TermName, flags: Long)(createFn: PolyMethodCreator): MethodSymbol = {
+    def newPolyMethod(typeParamCount: Int, owner: Symbol, name: TermName, flags: Long, enter: Boolean = true)(createFn: PolyMethodCreator): MethodSymbol = {
       val msym    = owner.newMethod(name.encode, NoPosition, flags)
       val tparams = msym.newSyntheticTypeParams(typeParamCount)
       val mtpe    = createFn(tparams) match {
         case (Some(formals), restpe) => MethodType(msym.newSyntheticValueParams(formals), restpe)
         case (_, restpe)             => NullaryMethodType(restpe)
       }
-
-      msym setInfoAndEnter genPolyType(tparams, mtpe) markAllCompleted
+      msym setInfo genPolyType(tparams, mtpe)
+      if (enter) msym.owner.info.decls enter msym
+      msym.markAllCompleted()
```

Comment:
See, I'd've added `sym.setInfoAndEnter(t, actuallyEnter: Boolean = true)` just to see who complained.

There is `newMethod` and `enterNewMethod` here. You could avoid the flag by following that time-honored pattern.

Actually there is `enterNewNullaryMethod`, and this rises to the same bar. No boolean flag!

I don't look at this code often. My sense is that it's like when they programmed by inserting jumpers in the earliest computers.